SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,685 @@
+﻿namespace SMLHelper.V2.Options
+{
+    using HarmonyLib;
+    using QModManager.API;
+    using SMLHelper.V2.Interfaces;
+    using SMLHelper.V2.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using UnityEngine;
+    using UWE;
+    using Logger = V2.Logger;
+#if SUBNAUTICA
+    using Text = UnityEngine.UI.Text;
+#elif BELOWZERO
+    using Text = TMPro.TextMeshProUGUI;
+#endif
+
+    /// <summary>
+    /// An internal derivative of <see cref="ModOptions"/> for use in auto-generating a menu based on attributes
+    /// declared in a <see cref="ConfigFile"/>.
+    /// </summary>
+    /// <typeparam name="TConfigFile">The type of the class derived from <see cref="ConfigFile"/> to use for
+    /// loading to/saving from disk.</typeparam>
+    internal class ConfigModOptions<TConfigFile> : ModOptions where TConfigFile : ConfigFile, new()
+    {
+        /// <summary>
+        /// The <typeparamref name="TConfigFile"/> <see cref="ConfigFile"/> instance related to this <see cref="ModOptions"/> menu.
+        /// </summary>
+        public TConfigFile Config { get; }
+
+        /// <summary>
+        /// Instantiates a new <see cref="ConfigModOptions{T}"/>, generating <see cref="ModOption"/>s by parsing the fields,
+        /// properties and methods declared in the class.
+        /// </summary>
+        public ConfigModOptions()
+            : base((typeof(TConfigFile).GetCustomAttributes(typeof(MenuAttribute), true).SingleOrDefault() as MenuAttribute)?.Name
+                  ?? QModServices.Main.GetMod(Assembly.GetCallingAssembly()).DisplayName)
+        {
+            // Instantiate and load the config
+            Config = new TConfigFile();
+            Config.Load();
+
+            menuAttribute = typeof(TConfigFile).GetCustomAttributes(typeof(MenuAttribute), true).SingleOrDefault() as MenuAttribute
+                ?? new MenuAttribute(Name);
+
+            // Process members
+            var bindingFlags = BindingFlags.Instance | BindingFlags.Public;
+            foreach (PropertyInfo property in typeof(TConfigFile).GetProperties(bindingFlags)
```

Comment:
If you're going to have a huge Linq statement to provide you the collection to iterate over, put that into a new variable outside so the main `foreach` condition doesn't span multiple lines.

Reply:
I see your point, I just like the semantic syntax of the declarative `foreach` in this case: it is very clear what is being looped over.

Moving that out into a new variable ahead of the `foreach` *might* allow that variable declaration to only span one line, but imo will sacrifice some readability in terms of being able to easily see "okay this is looping over the types where they are declared in a subclass of `ConfigFile` and where..."

And if your goal wasn't to make it so the variable declaration doesn't span only one line, just that the `foreach` condition doesn't... Isn't that making it harder to understand what's being looped over just for the sake of perceived tidyness?

Reply:
How about we just add braces `{ }` around the one line statement being executed? (to more cleanly separate what is part of the `foreach` declaration and what is the action)?

Reply:
Sure, I'm on board with that. Tbh, I usually always use braces - only haven't bc I have noticed in many places in SML/QMM code they aren't being used and figured that was preferred style so I started ditching them in some cases. If I was mistaken, I will start adding them back.

Reply:
I usually prefer no-braces on single-line execution statements, but if the control statement above it is going to span multiple lines, then I would stick with the braces there for clarity.

Reply:
Fair enough then :) ty for that clarification